src/components/pages/block/txs/index.tsx

```diff
@@ -0,0 +1,106 @@
+import { l2PublicClient } from "@/lib/chains";
+import Link from "next/link";
+import {
+  Table,
+  TableBody,
+  TableCaption,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from "@/components/ui/table";
+import { formatTimestamp } from "@/lib/utils";
+import { Card } from "@/components/ui/card";
+
+const BlckTxs = async ({ number }: { number: bigint }) => {
+  async function getGasUsed(hash: any, gasPrice: bigint) { 
+    let txr = l2PublicClient.getTransactionReceipt({hash: hash})
+    return (gasPrice * (await txr).gasUsed).toString()
+  }
+
+  const block = l2PublicClient.getBlock({
+    blockNumber: number,
+    includeTransactions: true,
+  });
+
+  const ts = formatTimestamp((await block).timestamp, false);
+
+  const transactions = (await block).transactions;
+
+  return (
+    <main className="flex flex-1 flex-col gap-4 p-4 md:gap-4 md:p-4">
+      <h2 className="mt-10 scroll-m-20 border-b pb-2 text-3xl font-semibold tracking-tight transition-colors first:mt-0">
+        Transactions
+      </h2>
+      <p>
+        For block{" "}
+        <Link
+          href={`/block/${number}`}
+          className="text-sm font-medium leading-none text-primary hover:brightness-150"
+        >
+          {number.toString()}
+        </Link>
+      </p>
+      <Card className="relative pl-8">
```

Comment:
This needs its own components, please refactor into `BlockTransactionsTable` taking a `transactions: Transaction[]` as props.
Keep the data fetching logic in this top-level Page component and the display logic in the Table component which receives properly computed tx values (separation of concerns).